packages/apputils-extension/src/statusbarplugin.ts

```diff
@@ -42,15 +43,30 @@ export const kernelStatus: JupyterFrontEndPlugin<IKernelStatusModel> = {
       sessionDialogs_ ?? new SessionContextDialogs({ translator });
     // When the status item is clicked, launch the kernel
     // selection dialog for the current session.
-    const changeKernel = async () => {
-      if (!item.model.sessionContext) {
-        return;
+    const changeKernel = async (event?: KeyboardEvent<HTMLImageElement>) => {
+      if (
+        event &&
+        (event.key === 'Enter' || event.key === 'Spacebar' || event.key === ' ')
+      ) {
+        event.preventDefault();
+        event.stopPropagation();
+        if (!item.model.sessionContext) {
+          return;
+        }
+        await sessionDialogs.selectKernel(item.model.sessionContext);
+      } else if (!event!.key) {
```

Comment:
This bit is hard to follow, could you decouple the key-handling logic (restore `changeKernel` as it was, and define a new function to be passed to `onKeyDown` that will invoke `changeKernel`)? Thank you!